src/Dialect/ONNX/ElementsAttr/ElementsAttrBuilder.cpp

```diff
@@ -450,6 +450,153 @@ std::vector<ElementsAttr> ElementsAttrBuilder::split(
   return results;
 }
 
+ElementsAttr ElementsAttrBuilder::concat(
+    ArrayRef<ElementsAttr> elms, unsigned axis) {
+  assert(elms.size() >= 1 && "concat tensors must be non-empty");
+
+  ElementsAttr first = elms.front();
+  auto firstShape = first.getType().getShape();
+  size_t rank = firstShape.size();
+  assert(axis < rank && "concat axis out of range");
+  if (elms.size() == 1)
+    return first;
+
+  // Check elms are compatible and construct outShape:
```

Comment:
It looks to me, the check here is redundant and perhaps overlapped with the concat op's verifier. It could be ok if the use of this function is out of scope of ONNXConcatOp, but I am not sure when it happens.

Reply:
`foldOrEmitONNXSplit/Transpose/Squeeze/UnsqueezeOp` functions was created at the time when we didn't have a way to fold ops. I believe the functions will be finally replaced by folding. So if we need the asserts for these functions, I am fine to keep them for a while.